DependencyInjection/CompilerPass/PluginPass.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Http\HttplugBundle\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Remove plugins if their dependencies are missing.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class PluginPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!($container->hasDefinition('logger') || $container->hasAlias('logger'))) {
+            $container->removeDefinition('httplug.plugin.logger');
+        }
+
+        if (!$container->hasDefinition('debug.stopwatch')) {
+            $container->removeDefinition('httplug.plugin.stopwatch');
+        }
+    }
```

Comment:
With this compiler pass we make sure that the exception messages makes sense. 

Before:
If logger plugin was or wasn't used and `logger` is not defined you would get an exception like "httplug.plugin.logger" requested an non existing service 'logger'". 

Now:
If logger plugin is used and `logger` is missing you will get an exception like "httplug.client.acme has requested an non existing service 'httplug.plugin.logger'"